gef.py

```diff
@@ -6772,11 +6784,11 @@ def do_invoke(self, _: List[str], **kwargs: Any) -> None:
 
         args : argparse.Namespace = kwargs["arguments"]
         if args.registers and args.registers[0]:
-            requested_regs = set(args.registers)
-            valid_regs = set(gef.arch.all_registers) & requested_regs
+            required_regs = set(args.registers)
+            valid_regs = [reg for reg in gef.arch.all_registers if reg in required_regs]
             if valid_regs:
```

Comment:
If the user provided a list of registers, none of which were valid this behaviour will result in it showing _all_ registers as it only updates `regs` (which is set to all above) if there are `valid_regs`. Is this what the user would expect? (Or does the `err` for `invalid_regs` below just stop it showing anything at all?)